File: app/src/types/Fruit.ts
export interface Fruit {
  id: number;
  positionY: number;
  positionX: number;
}

export function isFruit(value: unknown): value is Fruit {
  return (
    typeof value === 'object' &&
    value !== null &&
    typeof (value as Fruit).id === 'number' &&
    typeof (value as Fruit).positionY === 'number' &&
    typeof (value as Fruit).positionX === 'number'
  );
}

File: api/src/main/java/com/superdoodle/api/model/Fruit.java
package com.superdoodle.api.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.UUID;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class Fruit {
    private String id;

    private Integer positionY;

    private Integer positionX;

    public Fruit(Game game) {
        this.id = UUID.randomUUID().toString();
        this.positionX = (int) (Math.random() * game.getWidth());
        this.positionY = (int) (Math.random() * game.getHeight());
    }
}
